### PR TITLE
Remove usage of initializerToExpression from parse.d

### PR DESCRIPTION
**Pull request by RazvanN7**
`initializerToExpression` does a lot of type processing (which requires dependencies used at semantic time) so there is no need to be used in the parser. This further separates the parser from the semantic methods and leads the way to getting rid of the duplication in ASTBase

### CODE REVIEW

File: src/dmd/parse.d [hunk 1]
```diff
@@ -6488,8 +6488,11 @@ final class Parser(AST) : Lexer
                     value = parseInitializer();
                     if (token.value == TOK.colon)
                     {
+
```
Comment (wilzbach) [inline src/dmd/parse.d]: unrelated

**Review by jacob-carlborg — APPROVED** [review]
